code/game/turfs/initialization/dirty.dm

```diff
@@ -0,0 +1,46 @@
+/datum/turf_initializer/dirty/initialize(var/turf/simulated/T)
+	if(T.density)
+		return
+	// Quick and dirty check to avoid placing things inside windows
+	if(locate(/obj/structure/grille, T) || locate(/obj/structure/window_frame, T) || locate(/obj/structure/window/full, T))
+		return
+	//Don't place on openspace!
+	if(T.is_open())
+		return
+	//Dont place on unsimulated!
+	if(istype(T, /turf/unsimulated))
+		return
+
+	var/cardinal_turfs = T.CardinalTurfs()
+
+	T.dirt = rand(10, 50) + rand(0, 50)
+	// If a neighbor is dirty, then we get dirtier.
+	var/how_dirty = dirty_neighbors(cardinal_turfs)
+	for(var/i = 0; i < how_dirty; i++)
+		T.dirt += rand(0,10)
+	T.update_dirt()
+
+	if(prob(25))	// Keep in mind that only "corners" get any sort of web
+		attempt_web(T, cardinal_turfs)
```

Comment:
Old code, the proc accepts 1 argument, passing 2 here. The second argument is then unused? Maybe it is better to revisit this algorithm.